### ISSUE TITLE
plugin breaks installation of source tar.gz with pip on deployment host

### ISSUE THREAD

**Issue by seebi** [mtkennerly/poetry-dynamic-versioning#90]
Given this workflow:

- `poetry build` creates the source tar.gz (nicely versioned with poetry-dynamic-versioning)
- scp tar.gz file to deployment host
- `pip install ...tar.gz` on deployment host

we get errors such as `RuntimeError: This does not appear to be a Git project` or `RuntimeError: Unable to detect version control system.` during the installation.

I did not looked into the code (yet) so I can only guess here.

My expectations on this are:

- during the build, the version is evaluated and changed in the `pyproject.toml` (this seems to be true)
- during installation, the plugin does nothing at all (at this moment, no repo is there, so evaluating a version is not possible anyhow)

I see two options to do this:

- we define that poetry-dynamic-versioning acts ONLY in case, the  pyproject version is set to `0.0.0` (or any other documented value)
- during build, poetry-dynamic-versioning disables itself, changing not only `version` but also `enable=False` in the `[tool.poetry-dynamic-versioning]` section

btw. deployment host is debian bullseye docker container, so nothing special.

**Comment by seebi** [comment]
After looking briefly on the code - the first option seems to be a low hanging fruit by adding another return path here?

https://github.com/mtkennerly/poetry-dynamic-versioning/blob/master/poetry_dynamic_versioning/patch.py#L47

**Comment by mtkennerly** [comment]
The second option is how it works right now (or at least how it should):

https://github.com/mtkennerly/poetry-dynamic-versioning/blob/f1f56a01db2c8d909c7bb42838634e31e39517e9/poetry_dynamic_versioning/__init__.py#L258-L262

I don't immediately see why this would be misbehaving, but I'll do some testing and see what I can find. Are you using the latest version (v0.18.0)?

**Comment by mtkennerly** [comment]
Found the root cause (it was a silly mistake on my part) and just published [v0.19.0](https://github.com/mtkennerly/poetry-dynamic-versioning/releases/tag/v0.19.0) to fix it. Please let me know if you see any further issues.

**Comment by seebi** [comment]
I can confirm that the update solved the problem. Thanks a lot ...

```
$ poetry self remove "poetry-dynamic-versioning[plugin]"
...
$ poetry self add "poetry-dynamic-versioning[plugin]"
Using version ^0.19.0 for poetry-dynamic-versioning
...
```
